### PR TITLE
Use TValue as the generic type parameter

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/AspNetCore/issues/12926

### CODE REVIEW

File: src/Components/test/E2ETest/ServerExecutionTests/InteropReliabilityTests.cs [hunk 1]
```diff
@@ -80,7 +80,7 @@ public async Task CannotInvokeNonExistingMethods()
             await ValidateClientKeepsWorking(Client, batches);
         }
 
-        [Fact]
+        [Fact(Skip = "https://github.com/aspnet/AspNetCore/issues/12940")]
```
Comment (SteveSandersonMS) [inline src/Components/test/E2ETest/ServerExecutionTests/InteropReliabilityTests.cs]: The whole class is marked as `[Flaky]` currently. Do we also need to put `Skip` on individual tests?

**Review by SteveSandersonMS — APPROVED** [review]
Thanks for sorting this out!

As far as I can see there are no behavior changes here, just renames and moving code into separate files. If I missed any behavior change, please let me know.

**Comment by pranavkm** [comment]
> As far as I can see there are no behavior changes here, just renames and moving code into separate files. If I missed any behavior change, please let me know.

Yup, that's correct. I didn't realize the test class was skipped in entirety, I'll make a note in the tracking issue to remove the skip attributes.

**Comment by lpm-nova** [comment]
**Warning:** I am not a smart person. I'm sure this would normally be a RTFM situation. I... I just don't know if/where there's a manual for this at this time.

Using preview 8's InputSelect that contains the "T" parameter set to "int" I receive the runtime exception:
> Microsoft.AspNetCore.Components.Forms.InputSelect`1[System.Int32] does not support the type 'System.Int32'.

The same exception is displayed if I remove the T parameter.

Pulling the [source](https://github.com/aspnet/AspNetCore/blob/8c02467b4a218df3b1b0a69bceb50f5b64f482b1/src/Components/Web/src/Forms/InputSelect.cs), updated Aug 9th, I created a local duplicate class named InputSelectOfType<TValue>, changing only the name of the class. I received the same error when using TValue in the element and also when omitting it. ex:

`<InputSelectOfType TValue="int" id="certification-id" @bind-Value="@Question.CertificationId">`

If I change line 41 of  InputSelectOfType.cs (which is a direct copy of InputSelect.cs) from `else if (typeof(TValue).IsEnum)` to    `else` then I don't see errors.

**Comment by pranavkm** [comment]
@lpm-nova could you file a new issue with a scenario that reproduces the problem?

**Comment by lpm-nova** [comment]
It looks like my problem is a duplicate of [this issue](https://github.com/aspnet/AspNetCore/issues/11181). Altering line 41 works for `int` and that's about it.

I'll take Sanderson's advice and just override the method in a new class.

Sorry to waste your time.

**Comment by otaskiran** [comment]
Hi,
I have same int issue on 3.1.101 and newest preview sdk versions. How I can resolve it?

[2020-02-06T00:23:11.950Z] Error: System.InvalidOperationException: Microsoft.AspNetCore.Components.Forms.InputSelect`1[System.Int32] does not support the type 'System.Int32'.
   at Microsoft.AspNetCore.Components.Forms.InputSelect`1.TryParseValueFromString(String value, TValue& result, String& validationErrorMessage)
   at Microsoft.AspNetCore.Components.Forms.InputBase`1.set_CurrentValueAsString(String value)
   at Microsoft.AspNetCore.Components.Forms.InputSelect`1.<BuildRenderTree>b__4_0(String __value)
   at Microsoft.AspNetCore.Components.EventCallbackFactoryBinderExtensions.<>c__DisplayClass22_0`1.<CreateBinderCore>b__0(ChangeEventArgs e)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle)

**Comment by pranavkm** [comment]
@otaskiran support for `int` data types to work with `InputSelect` is being tracked by - https://github.com/dotnet/aspnetcore/issues/18305. You'd have to author a custom component to work around this issue at this point.

**Pull request merged**